src/main/java/io/github/debuggyteam/architecture_extensions/api/BlockGroup.java

```diff
@@ -31,13 +32,29 @@ public Iterator<BlockGroup.GroupedBlock> iterator() {
 		return groupedBlocks.iterator();
 	}
 
-	public record GroupedBlock(Identifier id, Supplier<Block> baseBlock, TextureConfiguration textureConfiguration, RecipeConfigurator recipeConfigurator, MapColor mapColor) {
+	public record GroupedBlock(Identifier id, Supplier<Block> baseBlock, TextureConfiguration textureConfiguration, RecipeConfigurator recipeConfigurator, Optional<MapColor> mapColor) {
 		public GroupedBlock(String id, Supplier<Block> baseBlock, TextureConfiguration textureConfiguration, RecipeConfigurator recipeConfigurator, MapColor mapColor) {
-			this(new Identifier(id), baseBlock, textureConfiguration, recipeConfigurator, mapColor);
+			this(new Identifier(id), baseBlock, textureConfiguration, recipeConfigurator, Optional.of(mapColor));
 		}
 		
 		public GroupedBlock(String id, Block baseBlock, TextureConfiguration textureConfiguration, RecipeConfigurator recipeConfigurator, MapColor mapColor) {
-			this(new Identifier(id), ()->baseBlock, textureConfiguration, recipeConfigurator, mapColor);
+			this(new Identifier(id), ()->baseBlock, textureConfiguration, recipeConfigurator, Optional.of(mapColor));
```

Comment:
Nitpicky but please put spaces between stuff such as `->`

Reply:
Let me look at the git blame really quick, hmmm it looks like it was falk who added that line